src/util/file.c

```diff
@@ -358,6 +358,30 @@ int fileput(char *path, char *string) {
     return(0);
 }
 
+int fileputat(int fd, char *path, char *string) {
```

Comment:
And we can further extend this usage of file descriptors to other places in the program, letting us remove our dependence on path name strings, and switch what we can to file descriptors. This also has the added benefit of removing pathways for attempting to attack singularity and escalate